pilot/pkg/xds/delta.go

```diff
@@ -350,32 +351,30 @@ func (s *DiscoveryServer) shouldRespondDelta(con *Connection, request *discovery
 	// the ack details and respond if there is a change in resource names.
 	con.proxy.Lock()
 	previousResources := con.proxy.WatchedResources[request.TypeUrl].ResourceNames
-	deltaResources := deltaWatchedResources(previousResources, request)
 	con.proxy.WatchedResources[request.TypeUrl].NonceAcked = request.ResponseNonce
 	con.proxy.WatchedResources[request.TypeUrl].NonceNacked = ""
-	con.proxy.WatchedResources[request.TypeUrl].ResourceNames = deltaResources
-	con.proxy.Unlock()
 
-	oldAck := listEqualUnordered(previousResources, deltaResources)
+	if !isWildcardTypeURL(request.TypeUrl) {
+		deltaResources := deltaWatchedResources(previousResources, request)
+		con.proxy.WatchedResources[request.TypeUrl].ResourceNames = deltaResources
+	}
+	if isWildcardTypeURL(request.TypeUrl) && (len(request.ResourceNamesSubscribe) > 0 || len(request.ResourceNamesUnsubscribe) > 0) {
+		panic(fmt.Sprintf("invalid assumption on %+v", request))
```

Comment:
should either guard with the `UNSAFE_PILOT_ENABLE_RUNTIME_ASSERTIONS` or remove. We don't want to panic in prod code.

Reply:
done